### DISCUSSION TITLE
Passing only some filters in queryString throws Exception "Undefined Index" for the excluded filters

### DISCUSSION THREAD

**Discussion by nicolaferrara82** [livewire/livewire#4789] [7. Bugs (archived)] [answered]
Clone my fresh repo https://github.com/nicolaferrara82/bug-livewire

Create some fake Posts;

**First Method to Reproduce**
go to url:
`/?filters[category]=one`

and will throw an Exception "Undefined index: search".

**Second Method**
this somehow leads back to the first bug

go to home page, then you will see the populated querystring
`?filters[search]=&filters[category]=`

select one category in list and hit enter in the address bar, empty filter search will be removed from querystring.
Now hit enter one more time or refresh the page.

Livewire component PostIndex:

```php
<?php

namespace App\Http\Livewire;

use App\Models\Post;
use Livewire\Component;

class PostIndex extends Component
{
    public $filters= [
        'search' => '',
        'category' => ''
    ];

    protected $queryString = ['filters'];

    public function resetAllFilters()
    {
        $this->reset('filters');
    }

    public function render()
    {
        $posts = Post::query()
            ->when($this->filters['search'], fn($query, $value) => $query->where('title', 'LIKE', '%' . $value . '%'))
            ->when($this->filters['category'], fn($query, $value) => $query->where('category', $value))
            ->get();
    
        return view('livewire.post-index', [
            'posts' => $posts,
            'categories' => ['one', 'two']
        ]);
    }
}
```

Livewire version: v2.10.4
Laravel version: v8.83.4

**Comment by nicolaferrara82** [comment, marked as answer]
I think the bug was is in SupportBrowserHistory constructor at "'component.hydrate.initial'" listener.

`$component->$property` is set to `$decoded === null ? $fromQueryString : $decoded;`

In this situation if $component->$property is an array, all preset keys are deleted.

Changing code with:

```php
$result = $decoded === null ? $fromQueryString : $decoded;
$component->$property = is_array($component->$property) ? array_merge($component->$property, $result) : $result;
```
seems to fix the bug.

**Reply by joshhanley** [reply]
@nicolaferrara82 any chance you can submit a PR for this with tests if you think it's a bug? 🙂 Thanks!
